lib/modules/datasource/go/base.ts

```diff
@@ -149,6 +149,19 @@ export class BaseGoDatasource {
 
       const registryUrl = `${parsedUrl.protocol}//${parsedUrl.host}`;
 
+      if (goModule.includes('.git')) {
```

Comment:
Is it for any includes or can be narrowed to `endsWith('.git')`?

Reply:
It is for any, as we could have `path/to/my/module.git/v3` (denoting version 3 of the go module at `path/to/my/module`). Go always suffixes the root module path with the major version (except v0/v1).
It might be possible to go with something like `/\.git(\/v\d+)?/` if additional safety is required?

Reply:
It's good to avoid false positives. Maybe `/\.git($|\/)` ? i.e. `.git` followed either by end of string or forward slash?

Reply:
Sounds reasonable, I updated both places and added another test for verifying the correct behaviour